File: packages/sdk/csharp/packages/core/FieldOps.cs
// EdgeBase C# SDK — Atomic field operation helpers.
// Usage:
//   await ref.UpdateAsync(id, new Dictionary<string, object?> {
//       { "views", FieldOps.Increment(1) },
//       { "temp",  FieldOps.DeleteField() },
//   });

namespace EdgeBase
{
    /// <summary>Atomic field operation markers for update methods.</summary>
    public static class FieldOps
    {
        /// <summary>
        /// Increment a numeric field atomically.
        /// Server: field = COALESCE(field, 0) + value.
        /// </summary>
        public static System.Collections.Generic.Dictionary<string, object?> Increment(double value = 1)
        {
            return new System.Collections.Generic.Dictionary<string, object?>
            {
                { "$op", "increment" },
                { "value", value }
            };
        }

        /// <summary>
        /// Delete a field (set to NULL).
        /// Server: field = NULL.
        /// </summary>
        public static System.Collections.Generic.Dictionary<string, object?> DeleteField()
        {
            return new System.Collections.Generic.Dictionary<string, object?>
            {
                { "$op", "deleteField" }
            };
        }
    }
}

File: packages/sdk/php/packages/core/src/FieldOps.php
<?php
/**
 * Atomic field operation helpers.
 *
 * Usage:
 *   $ref->update($id, [
 *       'views' => FieldOps::increment(1),
 *       'temp'  => FieldOps::deleteField(),
 *   ]);
 */

namespace EdgeBase\Core;

class FieldOps
{
    /**
     * Increment a numeric field atomically.
     * Server: field = COALESCE(field, 0) + $value.
     *
     * @param int|float $value Amount to increment (negative to decrement).
     * @return array{$op: string, value: int|float}
     */
    public static function increment(int|float $value = 1): array
    {
        return ['$op' => 'increment', 'value' => $value];
    }

    /**
     * Delete a field (set to NULL).
     * Server: field = NULL.
     *
     * @return array{$op: string}
     */
    public static function deleteField(): array
    {
        return ['$op' => 'deleteField'];
    }
}
